### PR TITLE
[MappingApplication] Fixing corner case from Nearest Neighbour

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

Fixes bug found in https://github.com/KratosMultiphysics/Kratos/pull/10581

Original code from @philbucher. I just fixed compilation and updated the results in tests. The modification consists in averaging values from nodes with the same distance (this rarely happens, but it may happen)

**🆕 Changelog**

- [Using all neighbors with closest distance](https://github.com/KratosMultiphysics/Kratos/commit/de9565749fcbb6ea940d2687427f02a88f77b821)
- [Fix compilation NN tests](https://github.com/KratosMultiphysics/Kratos/commit/244614520f9338176b2ec02876cb65801ff618bb)
- [Update solutions tests](https://github.com/KratosMultiphysics/Kratos/commit/3ad6c533ba4302c32b256ed69a2dbf52c70439c6)
- [Update corner case](https://github.com/KratosMultiphysics/Kratos/commit/ad4e23308c0ba850109ca16a8a8e8fb7bc344008)
- [Fix CPP test](https://github.com/KratosMultiphysics/Kratos/pull/10612/commits/d557e4c24ba35e490091a83df8c686eddb28f534)

### CODE REVIEW

**Comment by loumalouomega** [comment]
Huuum, Ci is failing, curious, I run all the tests....

**Comment by philbucher** [comment]
Cool thanks!

Please wait with merging, I will add a few more tests in the next days to cover the new developments

**Comment by loumalouomega** [comment]
@philbucher The barycentric tests are failing despite I have updated the results. Apparently, is not OMP. I cannot replicate this, I will try with another machine, but I don't like how this looks

**Comment by loumalouomega** [comment]
> @philbucher The barycentric tests are failing despite I have updated the results. Apparently, is not OMP. I cannot replicate this, I will try with another machine, but I don't like how this looks

Good news, the problem was that I was using an ARM64 machine and the results obtained were different. I reverted the result change, and now it passes in X86_64. Also, bad news, for some reason there are certain tests failing in ARM64, which a priori is not a good thing.

**Comment by loumalouomega** [comment]
@philbucher feel free to approve your own patch

File: applications/MappingApplication/tests/cpp_tests/test_nearest_neighbor_aux_classes.cpp [hunk 1]
```diff
@@ -146,9 +146,9 @@ KRATOS_TEST_CASE_IN_SUITE(NearestNeighborInterfaceInfo_MatchingNeighborFound, Kr
 
     KRATOS_CHECK(nearest_neighbor_info.GetLocalSearchWasSuccessful());
 
-    int found_id;
+    std::vector<int> found_id(1);
     nearest_neighbor_info.GetValue(found_id, MapperInterfaceInfo::InfoType::Dummy);
-    KRATOS_CHECK_EQUAL(found_id, expected_id_found);
+    KRATOS_CHECK_EQUAL(found_id[0], expected_id_found);
```
Comment (philbucher) [inline applications/MappingApplication/tests/cpp_tests/test_nearest_neighbor_aux_classes.cpp:151]: can you please extend this test to make sure both found nodes are tested?
I.e. test the content of found_id and also the weights

then it is ready, sorry that I didn't get to it yet :/

Comment (loumalouomega) [reply applications/MappingApplication/tests/cpp_tests/test_nearest_neighbor_aux_classes.cpp:151]: Test added

**Review by philbucher — APPROVED** [review]
thx

**Comment by loumalouomega** [comment]
FYI @pooyan-dadvand

**Pull request merged**